vyper/semantics/validation/local.py

```diff
@@ -493,50 +496,53 @@ class _LocalExpressionVisitor(VyperNodeVisitorBase):
     ignored_types = (vy_ast.Constant, vy_ast.Name)
     scope_name = "function"
 
-    def visit_Attribute(self, node: vy_ast.Attribute) -> None:
+    def visit(self, node, type_=None):
+        super().visit(node, type_)
+
+    def visit_Attribute(self, node: vy_ast.Attribute, type_: OptionalType) -> None:
         self.visit(node.value)
         _validate_msg_data_attribute(node)
         _validate_address_code_attribute(node)
 
-    def visit_BinOp(self, node: vy_ast.BinOp) -> None:
-        self.visit(node.left)
-        self.visit(node.right)
+    def visit_BinOp(self, node: vy_ast.BinOp, type_: OptionalType) -> None:
+        self.visit(node.left, type_)
+        self.visit(node.right, type_)
 
-    def visit_BoolOp(self, node: vy_ast.BoolOp) -> None:
+    def visit_BoolOp(self, node: vy_ast.BoolOp, type_: OptionalType) -> None:
         for value in node.values:  # type: ignore[attr-defined]
             self.visit(value)
 
-    def visit_Call(self, node: vy_ast.Call) -> None:
+    def visit_Call(self, node: vy_ast.Call, type_: OptionalType) -> None:
         self.visit(node.func)
         for arg in node.args:
-            self.visit(arg)
+            self.visit(arg, type_)
```

Comment:
this line looks wrong. but maybe, these visitors should not have an optional type anyways? as no functions in this visitor actually use it.